### DISCUSSION TITLE
Related-Values issue + Vue dependency issues

### DISCUSSION THREAD

**Discussion by ernst86913** [directus/directus#5430] [Archive]
Latest committed updates from May 2nd.  

Tried everything I can:   Reset Node Environment - removed node directory, npm install,....

1) Related-Values Issue in the latest update....
@directus/app:  error  in ./src/displays/related-values/related-values.vue
@directus/app: Syntax Error: TypeError: Cannot read property 'parseComponent' of undefined

2) Vue dependency issue in the latest updates...  (vue   vs   vue-template-compiler)

@ multi ../node_modules/webpack-dev-server/client?http://127.0.0.1:8080&sockPath=/sockjs-node ../node_modules/webpack/hot/dev-server.js ./src/main.ts
@directus/app: INTERNAL ERROR(undefined,undefined) When you use `vue` option, make sure 
to install `vue-template-compiler`.
@directus/app: stack trace:
@directus/app: Error: When you use `vue` option, make sure to install `vue-template-compiler`.
@directus/app:     at Function.resolveScriptBlock (C:\Users\rer\Documents\Coding\directus\node_modules\fork-ts-checker-webpack-plugin\lib\VueProgram.js:194:19)
@directus/app:     at Object.host.getSourceFile (C:\Users\rer\Documents\Coding\directus\node_modules\fork-ts-checker-webpack-plugin\lib\VueProgram.js:106:45)
@directus/app:     at findSourceFileWorker (C:\Users\rer\Documents\Coding\directus\node_modules\typescript\lib\typescript.js:108111:29)
@directus/app:     at findSourceFile (C:\Users\rer\Documents\Coding\directus\node_modules\typescript\lib\typescript.js:108021:26)
@directus/app:     at processImportedModules (C:\Users\rer\Documents\Coding\directus\node_modules\typescript\lib\typescript.js:108393:25)
@directus/app:     at findSourceFileWorker (C:\Users\rer\Documents\Coding\directus\node_modules\typescript\lib\typescript.js:108160:17)
@directus/app:     at findSourceFile (C:\Users\rer\Documents\Coding\directus\node_modules\typescript\lib\typescript.js:108021:26)
@directus/app:     at processImportedModules (C:\Users\rer\Documents\Coding\directus\node_modules\typescript\lib\typescript.js:108393:25)
@directus/app:     at findSourceFileWorker (C:\Users\rer\Documents\Coding\directus\node_modules\typescript\lib\typescript.js:108160:17)
@directus/app:     at findSourceFile (C:\Users\rer\Documents\Coding\directus\node_modules\typescript\lib\typescript.js:108021:26)

**Comment by rijkvanzanten** [comment]
Is this when building from source? What version of NPM are you using? Did you try removing all node_modules folders and running install fresh from the root before building?

**Reply by ernst86913** [reply]
Yes - building from LATEST sources.

npm: '7.5.3',

Yes - that's what I tried.

**Reply by rijkvanzanten** [reply]
Is this when running `npm run dev`? That issue path `../node_modules/webpack/hot/dev-server.js` indicates to me that this wasn't on build 🤔

**Reply by ernst86913** [reply]
Ah - sorry - yes - was on npm run dev

**Reply by rijkvanzanten** [reply]
Just to confirm, are there any `node_modules` left under any of the packages themselves? Maybe try running `npx lerna clean -y` to delete all the sub-package-node_modules, and then reinstall the root ones 🤔 

The only thing I can think of that's causing this on your end is some version mismatch of `vue-template-compiler` somehow (not seeing it on my end!)

**Reply by ernst86913** [reply]
I now installed latest version of NPM... again deleted npm cache + node_modules... now I get different errors it seems:

directus: 16:13:05 ✨ Server started at port 8055
@directus/app:  ERROR  Failed to compile with 1 error16:14:22
@directus/app: This dependency was not found:
@directus/app: * date-fns/formatDistance in ./src/utils/localized-format-distance/localized-format-distance.ts        
@directus/app: To install it, you can run: npm install --save date-fns/formatDistance
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/displays/datetime/datetime.vue(10,10):
@directus/app: 10:10 Module '"date-fns"' has no exported member 'parseISO'.
@directus/app:      8 | import localizedFormatDistance from '@/utils/localized-format-distance';
@directus/app:      9 | import i18n from '@/lang';
@directus/app:   > 10 | import { parseISO, parse } from 'date-fns';
@directus/app:        |          ^
@directus/app:     11 |
@directus/app:     12 | export default defineComponent({
@directus/app:     13 |         props: {
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/displays/datetime/datetime.vue(39,56):
@directus/app: 39:56 Expected 1-2 arguments, but got 3.
@directus/app:     37 |                                 return parseISO(props.value);
@directus/app:     38 |                         } else if (props.type === 'dateTime') {
@directus/app:   > 39 |                                 return parse(props.value, "yyyy-MM-dd'T'HH:mm:ss", new Date());
@directus/app:        |                                                                                    ^
@directus/app:     40 |                         } else if (props.type === 'date') {
@directus/app:     41 |                                 return parse(props.value, 'yyyy-MM-dd', new Date());
@directus/app:     42 |                         } else if (props.type === 'time') {
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/displays/datetime/datetime.vue(41,45):
@directus/app: 41:45 Expected 1-2 arguments, but got 3.
@directus/app:     39 |                                 return parse(props.value, "yyyy-MM-dd'T'HH:mm:ss", new Date());
@directus/app:     40 |                         } else if (props.type === 'date') {
@directus/app:   > 41 |                                 return parse(props.value, 'yyyy-MM-dd', new Date());
@directus/app:        |                                                                         ^
@directus/app:     42 |                         } else if (props.type === 'time') {
@directus/app:     43 |                                 return parse(props.value, 'HH:mm:ss', new Date());
@directus/app:     44 |                         }
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/displays/datetime/datetime.vue(43,43):
@directus/app: 43:43 Expected 1-2 arguments, but got 3.
@directus/app:     41 |                                 return parse(props.value, 'yyyy-MM-dd', new Date());
@directus/app:     42 |                         } else if (props.type === 'time') {
@directus/app:   > 43 |                                 return parse(props.value, 'HH:mm:ss', new Date());
@directus/app:        |                                                                       ^
@directus/app:     44 |                         }
@directus/app:     45 |
@directus/app:     46 |                         return null;
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/interfaces/datetime/datetime.vue(61,10):       
@directus/app: 61:10 Module '"date-fns"' has no exported member 'formatISO'.
@directus/app:     59 | import formatLocalized from '@/utils/localized-format';
@directus/app:     60 | import { i18n } from '@/lang';
@directus/app:   > 61 | import { formatISO, parseISO, format, parse } from 'date-fns';
@directus/app:        |          ^
@directus/app:     62 | 
@directus/app:     63 | export default defineComponent({
@directus/app:     64 |         props: {
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/interfaces/datetime/datetime.vue(61,21):       
@directus/app: 61:21 Module '"date-fns"' has no exported member 'parseISO'.
@directus/app:     59 | import formatLocalized from '@/utils/localized-format';
@directus/app:     60 | import { i18n } from '@/lang';
@directus/app:   > 61 | import { formatISO, parseISO, format, parse } from 'date-fns';
@directus/app:        |                     ^
@directus/app:     62 |
@directus/app:     63 | export default defineComponent({
@directus/app:     64 |         props: {
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/interfaces/datetime/datetime.vue(122,58):      
@directus/app: 122:58 Expected 1-2 arguments, but got 3.
@directus/app:     120 |                                                return parseISO(props.value);
@directus/app:     121 |                                        } else if (props.type === 'dateTime') {
@directus/app:   > 122 |                                                return parse(props.value, "yyyy-MM-dd'T'HH:mm:ss", new Date());
@directus/app:         |
     ^
@directus/app:     123 |                                        } else if (props.type === 'date') {
@directus/app:     124 |                                                return parse(props.value, 'yyyy-MM-dd', new Date());
@directus/app:     125 |                                        } else if (props.type === 'time') {
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/interfaces/datetime/datetime.vue(124,47):      
@directus/app: 124:47 Expected 1-2 arguments, but got 3.
@directus/app:     122 |                                                return parse(props.value, "yyyy-MM-dd'T'HH:mm:ss", new Date());
@directus/app:     123 |                                        } else if (props.type === 'date') {
@directus/app:   > 124 |                                                return parse(props.value, 'yyyy-MM-dd', new Date());
@directus/app:         |                                                                                        ^     
@directus/app:     125 |                                        } else if (props.type === 'time') {
@directus/app:     126 |                                                return parse(props.value, 'HH:mm:ss', new Date());
@directus/app:     127 |                                        }
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/interfaces/datetime/datetime.vue(126,45):      
@directus/app: 126:45 Expected 1-2 arguments, but got 3.
@directus/app:     124 |                                                return parse(props.value, 'yyyy-MM-dd', new Date());
@directus/app:     125 |                                        } else if (props.type === 'time') {
@directus/app:   > 126 |                                                return parse(props.value, 'HH:mm:ss', new Date());
@directus/app:         |                                                                                      ^       
@directus/app:     127 |                                        }
@directus/app:     128 |
@directus/app:     129 |                                        return null;
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/utils/get-date-fns-locale/get-date-fns-locale.ts(3,51):
@directus/app: 3:51 Cannot find name 'Locale'.
@directus/app:     1 | import { i18n } from '@/lang';
@directus/app:     2 |
@directus/app:   > 3 | export async function getDateFNSLocale(): Promise<Locale> {
@directus/app:       |                                                   ^
@directus/app:     4 |  const lang = i18n.locale;
@directus/app:     5 |
@directus/app:     6 |  const localesToTry = [lang, lang.split('-')[0], 'en-US'];
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/utils/localized-format-distance/localized-format-distance.ts(2,36):
@directus/app: 2:36 Cannot find module 'date-fns/formatDistance' or its corresponding type declarations.
@directus/app:     1 | import getDateFNSLocale from '@/utils/get-date-fns-locale';
@directus/app:   > 2 | import formatDistanceOriginal from 'date-fns/formatDistance';
@directus/app:       |                                    ^
@directus/app:     3 |
@directus/app:     4 | type LocalizedFormatDistance = (...a: Parameters<typeof formatDistanceOriginal>) => Promise<string>;
@directus/app:     5 |
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/utils/localized-format-distance/localized-format-distance.ts(8,3):
@directus/app: 8:3 Spread types may only be created from object types.
@directus/app:      6 | export const localizedFormatDistance: LocalizedFormatDistance = async (date, baseDate, options): Promise<string> => {
@directus/app:      7 |         return formatDistanceOriginal(date, baseDate, {
@directus/app:   >  8 |                 ...options,
@directus/app:        |                 ^
@directus/app:      9 |                 locale: await getDateFNSLocale(),
@directus/app:     10 |         });
@directus/app:     11 | };
@directus/app: Version: typescript 4.2.4
@directus/app: Time: 47826ms
@directus/app:  WAIT  Compiling...16:14:23
@directus/app: Error from chokidar (C:\): Error: EBUSY: resource busy or locked, lstat 'C:\hiberfil.sys'
@directus/app:  ERROR  Failed to compile with 1 error16:14:46
@directus/app: This dependency was not found:
@directus/app: * date-fns/formatDistance in ./src/utils/localized-format-distance/localized-format-distance.ts        
@directus/app: To install it, you can run: npm install --save date-fns/formatDistance
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/displays/datetime/datetime.vue(10,10):
@directus/app: 10:10 Module '"date-fns"' has no exported member 'parseISO'.
@directus/app:      8 | import localizedFormatDistance from '@/utils/localized-format-distance';
@directus/app:      9 | import i18n from '@/lang';
@directus/app:   > 10 | import { parseISO, parse } from 'date-fns';
@directus/app:        |          ^
@directus/app:     11 |
@directus/app:     12 | export default defineComponent({
@directus/app:     13 |         props: {
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/displays/datetime/datetime.vue(39,56):
@directus/app: 39:56 Expected 1-2 arguments, but got 3.
@directus/app:     37 |                                 return parseISO(props.value);
@directus/app:     38 |                         } else if (props.type === 'dateTime') {
@directus/app:   > 39 |                                 return parse(props.value, "yyyy-MM-dd'T'HH:mm:ss", new Date());
@directus/app:        |                                                                                    ^
@directus/app:     40 |                         } else if (props.type === 'date') {
@directus/app:     41 |                                 return parse(props.value, 'yyyy-MM-dd', new Date());
@directus/app:     42 |                         } else if (props.type === 'time') {
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/displays/datetime/datetime.vue(41,45):
@directus/app: 41:45 Expected 1-2 arguments, but got 3.
@directus/app:     39 |                                 return parse(props.value, "yyyy-MM-dd'T'HH:mm:ss", new Date());
@directus/app:     40 |                         } else if (props.type === 'date') {
@directus/app:   > 41 |                                 return parse(props.value, 'yyyy-MM-dd', new Date());
@directus/app:        |                                                                         ^
@directus/app:     42 |                         } else if (props.type === 'time') {
@directus/app:     43 |                                 return parse(props.value, 'HH:mm:ss', new Date());
@directus/app:     44 |                         }
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/displays/datetime/datetime.vue(43,43):
@directus/app: 43:43 Expected 1-2 arguments, but got 3.
@directus/app:     41 |                                 return parse(props.value, 'yyyy-MM-dd', new Date());
@directus/app:     42 |                         } else if (props.type === 'time') {
@directus/app:   > 43 |                                 return parse(props.value, 'HH:mm:ss', new Date());
@directus/app:        |                                                                       ^
@directus/app:     44 |                         }
@directus/app:     45 |
@directus/app:     46 |                         return null;
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/interfaces/datetime/datetime.vue(61,10):       
@directus/app: 61:10 Module '"date-fns"' has no exported member 'formatISO'.
@directus/app:     59 | import formatLocalized from '@/utils/localized-format';
@directus/app:     60 | import { i18n } from '@/lang';
@directus/app:   > 61 | import { formatISO, parseISO, format, parse } from 'date-fns';
@directus/app:        |          ^
@directus/app:     62 |
@directus/app:     63 | export default defineComponent({
@directus/app:     64 |         props: {
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/interfaces/datetime/datetime.vue(61,21):       
@directus/app: 61:21 Module '"date-fns"' has no exported member 'parseISO'.
@directus/app:     59 | import formatLocalized from '@/utils/localized-format';
@directus/app:     60 | import { i18n } from '@/lang';
@directus/app:   > 61 | import { formatISO, parseISO, format, parse } from 'date-fns';
@directus/app:        |                     ^
@directus/app:     62 |
@directus/app:     63 | export default defineComponent({
@directus/app:     64 |         props: {
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/interfaces/datetime/datetime.vue(122,58):      
@directus/app: 122:58 Expected 1-2 arguments, but got 3.
@directus/app:     120 |                                                return parseISO(props.value);
@directus/app:     121 |                                        } else if (props.type === 'dateTime') {
@directus/app:   > 122 |                                                return parse(props.value, "yyyy-MM-dd'T'HH:mm:ss", new Date());
@directus/app:         |
     ^
@directus/app:     123 |                                        } else if (props.type === 'date') {
@directus/app:     124 |                                                return parse(props.value, 'yyyy-MM-dd', new Date());
@directus/app:     125 |                                        } else if (props.type === 'time') {
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/interfaces/datetime/datetime.vue(124,47):      
@directus/app: 124:47 Expected 1-2 arguments, but got 3.
@directus/app:     122 |                                                return parse(props.value, "yyyy-MM-dd'T'HH:mm:ss", new Date());
@directus/app:     123 |                                        } else if (props.type === 'date') {
@directus/app:   > 124 |                                                return parse(props.value, 'yyyy-MM-dd', new Date());
@directus/app:         |                                                                                        ^     
@directus/app:     125 |                                        } else if (props.type === 'time') {
@directus/app:     126 |                                                return parse(props.value, 'HH:mm:ss', new Date());
@directus/app:     127 |                                        }
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/interfaces/datetime/datetime.vue(126,45):      
@directus/app: 126:45 Expected 1-2 arguments, but got 3.
@directus/app:     124 |                                                return parse(props.value, 'yyyy-MM-dd', new Date());
@directus/app:     125 |                                        } else if (props.type === 'time') {
@directus/app:   > 126 |                                                return parse(props.value, 'HH:mm:ss', new Date());
@directus/app:         |                                                                                      ^       
@directus/app:     127 |                                        }
@directus/app:     128 |
@directus/app:     129 |                                        return null;
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/utils/get-date-fns-locale/get-date-fns-locale.ts(3,51):
@directus/app: 3:51 Cannot find name 'Locale'.
@directus/app:     1 | import { i18n } from '@/lang';
@directus/app:     2 | 
@directus/app:   > 3 | export async function getDateFNSLocale(): Promise<Locale> {
@directus/app:       |                                                   ^
@directus/app:     4 |  const lang = i18n.locale;
@directus/app:     5 |
@directus/app:     6 |  const localesToTry = [lang, lang.split('-')[0], 'en-US'];
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/utils/localized-format-distance/localized-format-distance.ts(2,36):
@directus/app: 2:36 Cannot find module 'date-fns/formatDistance' or its corresponding type declarations.
@directus/app:     1 | import getDateFNSLocale from '@/utils/get-date-fns-locale';
@directus/app:   > 2 | import formatDistanceOriginal from 'date-fns/formatDistance';
@directus/app:       |                                    ^
@directus/app:     3 |
@directus/app:     4 | type LocalizedFormatDistance = (...a: Parameters<typeof formatDistanceOriginal>) => Promise<string>;
@directus/app:     5 |
@directus/app: Version: typescript 4.2.4
@directus/app: Time: 19750ms
@directus/app: ERROR in C:/Users/rer/Documents/Coding/directus/app/src/utils/localized-format-distance/localized-format-distance.ts(8,3):
@directus/app: 8:3 Spread types may only be created from object types.
@directus/app:      6 | export const localizedFormatDistance: LocalizedFormatDistance = async (date, baseDate, options): Promise<string> => {
@directus/app:      7 |         return formatDistanceOriginal(date, baseDate, {
@directus/app:   >  8 |                 ...options,
@directus/app:        |                 ^
@directus/app:      9 |                 locale: await getDateFNSLocale(),
@directus/app:     10 |         });
@directus/app:     11 | };

**Reply by ernst86913** [reply]
OK - will try the approach you suggested - thanks

**Reply by ernst86913** [reply]
Apologize for taking your time... the npx lerna clean -y  did it... thanks a lot for your help.

**Reply by rijkvanzanten** [reply]
Pfew glad to hear! Node module resolution can certainly get a little finicky at times.

**Reply by danieljwestman** [reply]
Sorry to say, but I have the same problem with building...

Totally clean install (RC62)

`npx lerna clean -y`

`npm install`

`npm update` (just to be sure)

`npm run build`

```
[...]

@directus/app@9.0.0-rc.62 build
vue-cli-service build

Starting type checking service...
Using 1 worker with 2048MB memory limit
 ERROR  Failed to compile with 1 error21:11:37
 error  in ./src/app.vue

Syntax Error: TypeError: Cannot read property 'parseComponent' of undefined

 @ ./src/main.ts 47:0-28 53:19-22
 @ multi ./src/main.ts

[...]
```

And when running `npm run dev`:

```
[...]

directus: 21:15:18 ✨ Server started at port 8055
@directus/app:  ERROR  Failed to compile with 1 error21:15:25
@directus/app:  error  in ./src/app.vue
@directus/app: Syntax Error: TypeError: Cannot read property 'parseComponent' of undefined
@directus/app:  @ ./src/main.ts 47:0-28 53:19-22
@directus/app:  @ multi ../node_modules/webpack-dev-server/client?http://127.0.0.1:8080&sockPath=/sockjs-node ../node_modules/webpack/hot/dev-server.js ./src/main.ts
@directus/app: INTERNAL ERROR(undefined,undefined) When you use `vue` option, make sure to install `vue-template-compiler`.
@directus/app: stack trace:
@directus/app: Error: When you use `vue` option, make sure to install `vue-template-compiler`.

[...]
```

Any ideas? 🤯

**Reply by rijkvanzanten** [reply]
Maybe try deleting the root node_modules as well after running `npx lerna clean -y`? I believe that only deletes the node_modules under the packages 🤔

**Reply by danieljwestman** [reply]
From Discord:

> AidenFoxx
> Did you do a lerna bootstrap?

And suddenly it works ✨

**Reply by rijkvanzanten** [reply]
5 points to @aidenfoxx 😄 

What I forgot to ask: What version of NPM are you using?

**Reply by danieljwestman** [reply]
npm 7.11.2
node 16.0.0

**Comment by ernst86913** [comment]
Got it working - thanks a lot 

LG,
Roman

Von meinem iPhone gesendet
